src/radical/pilot/raptor/worker_default.py

```diff
@@ -279,18 +279,29 @@ def _worker_proc(res_lock):
                 os.environ['CUDA_VISIBLE_DEVICES'] = \
                              ','.join(str(i) for i in task['slots']['gpus'])
 
+            sbox = task['description'].get('sandbox')
+            if not sbox:
+                sbox = self._sbox + '/' + task['uid']
+                task['description']['sandbox'] = sbox
```

Comment:
we should follow `sandbox` description in `TaskDescription`
```
That directory *MUST* be relative to the pilot sandbox
```
`sbox` should be taken from `task['task_sandbox_path']` (https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/session.py#L919, https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/tmgr/scheduler/base.py#L349)
```python
if task['description'].get('sandbox'):
    sbox = task['task_sandbox_path']
else:
    sbox = self._sbox + '/' + task['uid']
```
`self._sbox` is referred to Worker sandbox, right?

Reply:
> self._sbox is referred to Worker sandbox, right?

Indeed

> That directory *MUST* be relative to the pilot sandbox

Hmm, I don't think we actually need this limitation anymore - the handling of absolute paths should be fixed by now.  Well, apart from what I added in this PR :-/  So let me fix this...

Ack on the `session._get_task_sandbox` call, but we need to have that available on the agent side also for raptor submitted tasks.

Reply:
This is fixed now.  A side effect though is that the default sboxes are relative to the pilot, not the worker.  Which is consistent thoufg with other task which is good.